pkg/multicluster/reconcile/reconcile.go

```diff
@@ -39,14 +39,14 @@ func NewLoop(name string, cw multicluster.ClusterWatcher, resource ezkube.Object
 // AddCluster creates a reconcile loop for the cluster.
 func (r *clusterLoopRunner) AddCluster(ctx context.Context, cluster string, mgr manager.Manager) {
 	loopForCluster := reconcile.NewLoop(r.name+"-"+cluster, mgr, r.resource)
+
+	// Cleanup any potentially stale references to the cluster.
+	r.clusterLoops.remove(cluster)
+	r.reconcilers.unset(cluster)
```

Comment:
If we clean up the cluster loop and reconcile sets like this, it's possible for the cluster loop map in r.clusterLoops and cluster sets in r.reconcilers to grow indefintiely as clusters are removed from the system.

Fixes for this include the goroutine approach this replaced or adding an optional RemoveCluster function to the cluster handler.

Reply:
https://github.com/solo-io/skv2/pull/4/commits/17cf19cc0042d6d49e9e2a971410ccbea1550f4c